plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -124,7 +125,8 @@ private static boolean isTrackingIdDefined() {
         && Constants.ANALYTICS_TRACKING_ID.startsWith("UA-");
   }
 
-  private String getAnonymizedClientId() {
+  @VisibleForTesting
+  String getAnonymizedClientId() {
     String clientId = preferences.get(AnalyticsPreferences.ANALYTICS_CLIENT_ID, null);
     if (clientId == null) {
       clientId = UUID.randomUUID().toString();
```

Comment:
any thread safety issues here? Could we have analytics pinging from two separate operations at once? E.g. user starts a local server and then starts a deploy before the server has started up?